src/enzo/Grid_SchrodingerSolver.C

```diff
@@ -0,0 +1,350 @@
+    CellWidthTemp[dim][i] = float(a*CellWidth[dim][i]);
+  else
+    CellWidthTemp[dim][i] = 1.0;
+    }
+
+      /* Prepare Gravity. */
+
+  int GravityOn = 0, FloatSize = sizeof(float);
+    if (SelfGravity || UniformGravity || PointSourceGravity || DiskGravity || ExternalGravity )
+      GravityOn = 1;
+#ifdef TRANSFER
+    if (RadiationPressure)
+      GravityOn = 1;
+#endif  
+
+  /* Error Check */
+
+  for (i = 0; i < GridRank; i++)
+    if (GridDimension[i] > MAX_ANY_SINGLE_DIRECTION) {
+      ENZO_FAIL("Schrodinger_MAIN: A grid dimension is too long (increase max_any_single_direction.)\n");
+    }
+
+  /* Allocate temporary space for Zeus_Hydro. */
+
+  int size = GridDimension[0]*GridDimension[1]*GridDimension[2];
+
+  //float *repsi0 = new float[size];
+  //float *impsi0 = new float[size];
+  
+  /* Find fields: density, total energy, velocity1-3 and set pointers to them
+     Create zero fields for velocity2-3 for low-dimension runs because solver
+     assumes they exist. */
+
+  /*this->IdentifyPhysicalQuantities(DensNum, GENum, Vel1Num, Vel2Num, Vel3Num, TENum );
+  
+  d = BaryonField[DensNum];
+  e = BaryonField[TENum];
+  u = BaryonField[Vel1Num];
+  v = BaryonField[Vel2Num];
+  w = BaryonField[Vel3Num]; */
+
+  /*  Set grid start and end indicies */
+
+  /*is = GridStartIndex[0];
+  js = GridStartIndex[1];
+  ks = GridStartIndex[2];
+  ie = GridEndIndex[0];
+  je = GridEndIndex[1];
+  ke = GridEndIndex[2];*/
+
+  /* if use more ghost zones */  
+  /*ie = ie+max(0,is-3);
+  je = je+max(0,js-3);
+  ke = ke+max(0,ks-3);
+  is = is-max(0,is-3);
+  js = js-max(0,js-3);
+  ks = ks-max(0,ks-3);*/
+
+		 
+  /*   1) Add source terms */  
+    // get code units
+  float TemperatureUnits = 1, DensityUnits = 1, LengthUnits = 1,
+    VelocityUnits = 1, TimeUnits = 1;
+
+  if (QuantumGetUnits(&DensityUnits, &LengthUnits, &TemperatureUnits,
+         &TimeUnits, &VelocityUnits, Time) == FAIL) {
+    ENZO_FAIL("Error in GetUnits.");
+  }
+
+// set poteitial offset  
+int Offset[MAX_DIMENSION] = {0,0,0};
+for (int dim = 0; dim < GridRank; dim++) {
+    Offset[dim] = nint((CellLeftEdge[dim][0] -
+      GravitatingMassFieldLeftEdge[dim])/ CellWidth[dim][0]);
+    //printf("offset %d %d\n", dim , GravitatingMassFieldDimension[dim]);
+  }
+
+  // calculate hbar/m
+  float hmcoef = 5.9157166856e27*TimeUnits/pow(LengthUnits,2)/FDMMass;
+  //printf("hmcoef %f \n", hmcoef);
+
+  int RePsiNum, ImPsiNum, FDMDensNum;
+  RePsiNum = FindField(RePsi, FieldType, NumberOfBaryonFields);
+  ImPsiNum = FindField(ImPsi, FieldType, NumberOfBaryonFields);
+  FDMDensNum = FindField(FDMDensity, FieldType, NumberOfBaryonFields);
+
+  float *repsi, *impsi, *d;
+  repsi = BaryonField[RePsiNum];
+  impsi = BaryonField[ImPsiNum];
+  d = BaryonField[FDMDensNum];
+
+  // Add half gravity before the TVD-RK
+  if (GravityOn && (PotentialField != NULL) ){
+    if (nhy == 0 ){ // Add half step of potential at the beginning
+    if(SchrdingerAddPotential(repsi, impsi,
```

Comment:
Is anybody else bothered by the misspelling of "schrodinger"?  This is everywhere...

Reply:
Fixed!